linkerd/proxy/detect/src/lib.rs

```diff
@@ -17,41 +17,26 @@ pub trait Detect<T>: Clone {
     fn detect_peeked_prefix(&self, target: T, prefix: &[u8]) -> Self::Target;
 }
 
-#[derive(Debug, Clone)]
-pub struct Accept<D, A> {
-    detect: D,
-    accept: A,
-    peek_capacity: usize,
-}
-
-pub enum AcceptFuture<T, D, A>
-where
-    D: Detect<T>,
-    A: core::listen::Accept<(D::Target, BoxedIo)>,
-{
-    Accept(A::Future),
-    Detect {
-        detect: D,
-        accept: A,
-        inner: PeekAndDetect<T, D>,
-    },
-}
-
 pub enum PeekAndDetect<T, D: Detect<T>> {
-    // Waiting for accept to become ready.
+    // Waiting for the inner svc to become ready.
     Detected(Option<(D::Target, BoxedIo)>),
     // Waiting for the prefix to be read.
     Peek(Option<T>, Peek<BoxedIo>),
 }
 
-impl<D, A> Accept<D, A> {
+#[derive(Debug, Clone)]
+pub struct DetectProtocolLayer<D> {
+    detect: D,
+    peek_capacity: usize,
+}
+
+// === impl DetectProtoLayer ===
+impl<D> DetectProtocolLayer<D> {
     const DEFAULT_CAPACITY: usize = 8192;
 
-    /// Creates a new `Detect`.
-    pub fn new(detect: D, accept: A) -> Self {
-        Self {
+    pub fn new(detect: D) -> DetectProtocolLayer<D> {
+        DetectProtocolLayer {
```

Comment:
Generally, we prefer the `Self` form... minimizes churn if names change.
```suggestion
    pub fn new(detect: D) -> Self {
        Self {
```